src/Converter/LineItemsConverter.php

```diff
@@ -23,7 +36,7 @@ public function convert(OrderInterface $order): Collection
         foreach ($orderItems as $orderItem) {
             $variant = $orderItem->getVariant();
 
-            $lineItems->add(new LineItem(
+            $lineItems->add(new $this->className(
```

Comment:
Why not use factories for that?

> $lineItems->add($this->lineItemFactory->createWithData(

Reply:
You are right there should be a factory, but none was created before and since this PR is a little big, I wanted to focus it on fixing existing issues and not on creating new classes.